lib/google/protobuf/duration.ex

```diff
@@ -51,9 +51,17 @@ defimpl Protox.JsonMessageDecoder, for: Google.Protobuf.Duration do
 
   # -- Private
 
+  @dialyzer {:no_match, get_decimal: 1}
```

Comment:
I'm not familiar with this dialyzer directive, I guess it deactivates verification about pattern matching? And more specifically, why was it needed here? I can't see anything suspicious regarding types 🤔.

Reply:
your guess is correct, it disables pattern matching warning. Since the version is pinned to 1.9, I assume it picks up the types from 1.9 which never returns {decimal, binary}. Since we won't be able to make dialyzer happy here as the return type is different in both versions, I just disabled the specific error for this function only.


```
Total errors: 2, Skipped: 0, Unnecessary Skips: 0
done in 0m2.58s
lib/google/protobuf/duration.ex:64:pattern_match
The pattern can never match the type.

Pattern:
{_dec, <<>>}

Type:
:error

________________________________________________________________________________
lib/google/protobuf/duration.ex:65:pattern_match
The pattern can never match the type.

Pattern:
{__dec, __suffix}

Type:
:error

________________________________________________________________________________
done (warnings were emitted)
```

Reply:
Ok, thanks for the explanation! I guess it's OK to merge then ;-)